hyperspy/roi.py

```diff
@@ -322,11 +322,13 @@ def interactive(self, signal, navigation_signal="same", out=None,
             If not None, it will use 'out' as the output instead of returning
             a new Signal.
         """
-        if navigation_signal == "same":
+        if isinstance(navigation_signal, basestring) and navigation_signal == "same":
             navigation_signal = signal
         if navigation_signal is not None:
             if navigation_signal not in self.signal_map:
                 self.add_widget(navigation_signal)
+        if self.update not in signal.events.data_changed.connected:
+            signal.events.data_changed.connect(self.update, [])
```

Comment:
What is the purpose of this? The widget and ROIs should be unaffected by any changes to the data/axes as they operate in value space.